os/arch/arm/src/armv7-m/up_assert.c

```diff
@@ -440,11 +443,27 @@ static void _up_assert(int errorcode)
 		request_msg.cmd = BINMGR_FAULT;
 		request_msg.pid = getpid();
 
+		if (current_regs) {
+			tcb = sched_self();
+			sched_removereadytorun(tcb);
+#if CONFIG_RR_INTERVAL > 0
+			tcb->timeslice = 0;
+#endif
+			tcb->sched_priority = SCHED_PRIORITY_MIN;
+			sched_addreadytorun(tcb);
+		}
+
 		/* Send a message to fault manager with pid of the faulty task */
 		ret = mq_send(binmgr_mq, (const char *)&request_msg, sizeof(binmgr_request_t), BINMGR_FAULT_PRIO);
 		DEBUGASSERT(ret == 0);
+
+		if (current_regs) {
+			tcb = sched_self();
+			current_regs = tcb->xcp.regs;
+		}
```

Comment:
Why is this different with armv7-r change which you've modified?
```
		binmgr_mq = mq_open(BINMGR_REQUEST_MQ, O_WRONLY, 0666, 0);
		DEBUGASSERT(binmgr_mq != (mqd_t)ERROR);

		request_msg.cmd = BINMGR_FAULT;
		request_msg.pid = getpid();

		/* Check if the fault is due to data/prefetch/undef abort or
		   due to direct call to up_assert */

		if (current_regs) {

			/* if fault is due to any kind of abort,
			   enable the interrupts to schedule to next task */

			current_regs = NULL;
			irqenable();
		}

		/* Send a message to fault manager with pid of the faulty task */

		ret = mq_send(binmgr_mq, (const char *)&request_msg, sizeof(request_msg), BINMGR_FAULT_PRIO);
		DEBUGASSERT(ret == 0);

		exit(errorcode);
```

Reply:
The way of handling aborts are little different in R and M architectures. current_regs are managed in different way, so we have modified the changes to meet the M requirements.
We will modify the R implementation similar to the current M design and update in separate PR after testing.